src/discussions/topics/topic-group/TopicGroup.jsx

```diff
@@ -1,24 +1,68 @@
+import PropTypes from 'prop-types';
 import React from 'react';
+import { useParams } from 'react-router';
+import { Link } from 'react-router-dom';
+import { Routes } from '../../../data/constants';
 import Topic, { topicShape } from './topic/Topic';
 
-function TopicGroup({ id, name, topics }) {
+function TopicGroup({
```

Comment:
IMO it would be better to introduce a `FilteredTopicGroups` component which encapsulates the filtering, something like:
```js
const filterFun = filter ? tg => tg.name?.toLowerCase().includes(filter) : () => true;
topicGroups.filter(tg => filterFun(tg) || tg.children.filter(filterFun)).map(tg => <TopicGroup ...);
```